ROSTESTS-263. KmTest: Add 1 usage check and 1 cleanup

SergeGautherie opened the pull request:
## Purpose

JIRA issue: [ROSTESTS-263](https://jira.reactos.org/browse/ROSTESTS-263)

## Proposed changes

- kmtest.c: Add a check and report for "Too many arguments".
- support.c: Add a KmtDeleteService() call to KmtUnloadDriver(), Also add some result reports.

SergeGautherie commented:
Added `skip(FALSE, "KmtDeleteService('%ls') will not be called...\n", TestServiceName);`

SergeGautherie commented:
2 s/2017/2018/

SergeGautherie commented:
Added `ok(FALSE, "KmtCreateAndStartService ...` to be consistent, though this doesn't happen either.

SergeGautherie commented on modules/rostests/kmtests/kmtest/support.c:
```diff
@@ -159,8 +163,28 @@ KmtUnloadDriver(VOID)
+        ok(FALSE, "KmtStopService('%ls') failed! (Error = 0x%08lx)\n", TestServiceName, Error);
+
         // TODO
         __debugbreak();
```
Should I remove these `__debugbreak();`?
Otherwise, what are these TODOs intended to become?

learn-more replied:
I would assume this is something that is not finished, hence why someone wanted to break in here.

SergeGautherie replied:
Hopefully, @ThFabba, as author, shall answer these with more details.

learn-more commented on modules/rostests/kmtests/kmtest/kmtest.c:
```diff
@@ -1,8 +1,9 @@
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Kernel-Mode Test Suite Loader Application
+ * COPYRIGHT:   Copyright 2011 Thomas Faber <thomas.faber@reactos.org>
+ *              Copyright 2018 Serge Gautherie <reactos-git_serge_171003@gautherie.fr>
```
Adding 2 lines, and already adding copyright?

SergeGautherie replied:
Why not? I can remove this line if unwanted...

SergeGautherie commented:
Fixed a few additional comment copypastas.
Added a few minor improvements.
Extended test result output.

The latter prints KmTest Driver itself start/stop trace() for example, as "sc start/stop" does.

SergeGautherie commented:
Added `DPRINT("KmtCreateAndStartService ...`.

SergeGautherie commented:
Retrying failed c-i build.

SergeGautherie commented:
Retrying failed build.

SergeGautherie commented:
Unbitrotted license header.

SergeGautherie commented:
@ThFabba, could you update your 'Changes requested'?

ThFabba commented on modules/rostests/kmtests/kmtest/kmtest.c:
```diff
@@ -316,8 +317,11 @@ main(
-    if (ArgCount <= 1)
+    if (ArgCount != 2)
     {
+        if (ArgCount > 2)
```
Wine tests frequently accept additional arguments that can be parsed by the test. So far kmtests haven't needed this feature, but it can easily be added. Prohibiting it at this point does not seem useful.

ThFabba commented on modules/rostests/kmtests/kmtest/support.c:
```diff
@@ -141,15 +142,21 @@ KmtLoadDriver(
 
     if (Error)
     {
+        ok(FALSE, "KmtCreateAndStartService(\"%ls\") failed! (Error = 0x%08lx)\n", TestServiceName, Error);
```
This is not part of a test, using `ok` here seems inappropriate.

ThFabba commented on modules/rostests/kmtests/kmtest/support.c:
```diff
@@ -141,15 +142,21 @@ KmtLoadDriver(
     }
+    else
+    {
+        DPRINT("KmtCreateAndStartService(\"%ls\") succeeded\n", TestServiceName);
```
DPRINTs in a console application do not seem useful.

ThFabba commented on modules/rostests/kmtests/kmtest/support.c:
```diff
@@ -141,15 +142,21 @@ KmtLoadDriver(
+        ok(FALSE, "KmtCreateAndStartService(\"%ls\") failed! (Error = 0x%08lx)\n", TestServiceName, Error);
+
         // TODO
         __debugbreak();
```
The TODO means "properly handle failure here".
This may be as simple as printing an error message and returning.
Or if callers rely on this function always succeeding, it may require either a call to `exit` or refactoring this to return an error code and adding error handling in the callers.

ThFabba commented:
Superseded by f027384e481.
